cudax/include/cuda/experimental/__device/device_ref.cuh

```diff
@@ -119,6 +119,27 @@ public:
     return __name;
   }
 
+  // TODO not sure if this is the best name :(
+  //! @brief Queries if its possible for this device to directly access specified device's memory.
+  //!
+  //! If this function returns true, device supplied to this call can be passed into enable_peer_access
+  //! on memory resource or pool that manages memory on this device. It will make allocations from that
+  //! pool accessible by this device.
+  //!
+  //! @param __other_dev Device to query the peer access
+  //! @return true if its possible for this device to access the specified device's memory
+  bool can_peer_access_to(device_ref __other_dev) const
```

Comment:
How about `can_access_peer`

Reply:
I would be fine with it too, I guess the issue is that in *almost* all cases the peer access is symmetrical, but technically there could be cases where it's not. I tried to capture the direction in the name `dev1.can_peer_access_to(dev2)`, but I don't know if its a good idea.

Reply:
`dev1.has_peer_access_to(dev2)`? or just `dev1.is_peer_of(dev2)`?

Reply:
What complicates it further is that this API queries only the possibility of peer access, while the actual enablement is a separate API (cudaDeviceEnablePeerAccess). But maybe because we don't expose the device level enablement of peer access and instead do it on memory pool / memory resource basis, then it's fine and these two won't be confused?
(So capability query is done on device level, enablement is on memory_pool / memory_resource and type separation handles the semantic separation).

I also very much like how `is_peer_of` reads like, the main problem is that technically it's possible to have GPU1 with peer access to GPU2, but not the other way. Being a peer sounds like a symmetric relation, while "peer access to" less so?
I think I would lean towards "peer accessible" meaning a GPU can access another GPU memory and "peers" being both ways peer accessible GPUs, like in `get_peers`. Since this is a one way query it would stay as `peer_access_to` and I think `has` might be better than `can`. I will update the name.

Another problem is I am not sure if the current CUDA Programming Guide considers peer devices to be just any two devices or peer access capable devices. But maybe it doesn't stop as from committing to the above definition of "peer" devices.